### PR TITLE
DROOLS-2263 : Unexpected results in GDST when using enumerations with commas

### PR DESCRIPTION
**Pull request by Rikkola**
https://issues.jboss.org/browse/DROOLS-2263

The fix is different than the one for 6.5.x. Still doing the same thing in the end. Just on different locations.

Bundle:
https://github.com/kiegroup/kie-soup/pull/29
https://github.com/kiegroup/drools/pull/1727
https://github.com/kiegroup/drools-wb/pull/777

### CODE REVIEW

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/columns/ValueListUiColumn.java [hunk 1]
```diff
@@ -28,79 +28,104 @@
...
+                                       final GridBodyCellRenderContext context) {
+        t.setText(getLabel(value.toString()));
     }
 
+    private String getLabel(final String value) {
+        if (!isMultipleSelect) {
+            return getValue(value);
+        }
+        final StringBuilder sb = new StringBuilder();
+        final String[] values = value.split(",");
```
Comment (manstis) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/columns/ValueListUiColumn.java]: I know it's old code but should I be worried about the split here?

Comment (Rikkola) [reply drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/columns/ValueListUiColumn.java]: Double checking, perhaps the fix is not the best.

Comment (Rikkola) [reply drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/columns/ValueListUiColumn.java]: @manstis Ok, I think this is the best way to do it. Unless you know why the list is empty when the value is list. For example with a cell value "a,b" the a or b can not be found from the `valueListLookup` map.

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/utilities/ColumnUtilities.java [hunk 2]
```diff
@@ -142,7 +143,7 @@ public static boolean canAcceptOtherwiseValues(final BaseColumn column) {
     private String[] parseValueList(String fieldType,
                                     String valueList) {
 
-        final String[] values = valueList.split(",");
+        final String[] values = ListSplitter.split(valueList);
```
Comment (jomarko) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/utilities/ColumnUtilities.java]: Could we test this change in `org.drools.workbench.screens.guided.dtable.client.widget.table.utilities.ColumnUtilitiesGetValueListTest` class please?

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/columns/ValueListUiColumn.java [hunk 3]
```diff
@@ -28,79 +28,104 @@
...
+                                       final String value,
+                                       final GridBodyCellRenderContext context) {
+        t.setText(getLabel(value.toString()));
     }
 
+    private String getLabel(final String value) {
+        if (!isMultipleSelect) {
+            return getValue(value);
+        }
+        final StringBuilder sb = new StringBuilder();
```
Comment (jomarko) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/columns/ValueListUiColumn.java]: I believe we don't need the `StringBuilder sb` and we can implement the `for` cycle below as:
```java
Stream.of(values).collect(Collectors.joining(","));
```

Comment (Rikkola) [reply drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/columns/ValueListUiColumn.java]: Changing, the code is more clear this way.

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/utilities/ColumnUtilities.java (hunk 2 above)
Comment (Rikkola) [reply drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/utilities/ColumnUtilities.java]: Tried with a break point and this does get hit with several different setups where the character , is in the string.

**Comment by jomarko** [comment]
@manstis this is ready for merge.

Ensemble 1/4.

**Pull request merged**